js/customize-partial-refresh-settings.js

```diff
@@ -0,0 +1,128 @@
+/* global jQuery, _customizePartialRefreshSettings, JSON, _ */
+/* exported customizePartialRefreshSettings */
+var customizePartialRefreshSettings = ( function( $ ) {
+	'use strict';
+
+	var self, api = wp.customize;
+
+	self = {
+		setting: null
+	};
+
+	$.extend( self, _customizePartialRefreshSettings );
+
+	/**
+	 * Inject the functionality.
+	 */
+	self.init = function() {
+		api.bind( 'ready', function() {
+			self.onChangeSetting();
+		} );
+
+		api.bind( 'setting-partial', function( response ) {
+			var that = $( 'iframe' ).contents(),
+				value = $.trim( response );
+
+			$( self.setting.selector, that ).html( value );
```

Comment:
If the given selector matches no elements, then maybe this should also be a scenario where the preview refreshes?

In the case of sending messages, the preview would then need to do `wp.customize.preview.send( 'refresh' )` if it can't find the given selector, if the pane is doing something like this:

```js
wp.customize.previewer.send( 'selective-refresh', selector, response );
```

The preview can then do:

```js
wp.customize.preview.bind( 'selective-refresh', function( selector, response ) {
    var elements = jQuery( selector );
    if ( elements.length ) {
        elements.html( jQuery.trim( response ) );
    } else {
        wp.customize.preview.send( 'refresh' )
    }
} );
```